lib/katex.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const { renderToString: renderKatex } = require('katex');
+
+module.exports = hexo => {
+  return function katexFn(args, content) {
+    if (this.katex === false) return content;
+
+    const { options: globalCfg } = hexo.config.math.katex;
+    const { katex: fmCfg } = this;
+    const [jsonCfg] = args;
+    const argsCfg = jsonCfg ? JSON.parse(jsonCfg) : false;
+    let options = Object.assign({}, globalCfg);
+    if (fmCfg) options = Object.assign({}, options, fmCfg);
+    if (argsCfg) options = Object.assign({}, options, argsCfg);
```

Comment:
```suggestion
    if (fmCfg) options = { ...options, fmCfg };
    if (argsCfg) options = { ...options, argsCfg };
```

When the first parameter of `Object.assign` is an empty plain object, it will be slower than the spread syntax.